pandas/_libs/reduction.pyx

```diff
@@ -543,7 +545,10 @@ def apply_frame_axis0(object frame, object f, object names,
                 # `piece` might not have an index, could be e.g. an int
                 pass
 
-            results.append(piece)
+            if is_scalar(piece):
+                results.append(piece)
+            else:
```

Comment:
instead of doing this, just change the ``.copy()`` above and use ``copy.deepcopy``; I think we are somehow not copying correctly here

Reply:
I think we're only reaching the `.copy()` if `f` returns something that has an index.  In this case `piece` is itself an index so we hit the exception and never have the chance to copy.

Reply:
we almost always have a Series already, so this is double copying.

Reply:
Looking into how to avoid this; for some reason when I take out the first copy and rely instead on a `deepcopy` here I get a couple test failures.  Will work on seeing if I can resolve those.

Reply:
Can this check be merged into the try/except above?  e.g. it may be that the `is_scalar(piece)` cases are identically the `AttributeError` cases

Reply:
I don't think they are because of the example in the test case (an `Index` triggers the `AttributeError` but isn't a scalar).  It seems that we can move the `copy` into the `is_scalar` check and capture all the cases that need copying (and copy only once per @jreback 's point).

Reply:
OK, sounds like you've looked at this pretty carefully with an eye to minimizing performance impact